src/Adapter/Product/CommandHandler/SetProductSuppliersHandler.php

```diff
@@ -0,0 +1,256 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\CommandHandler\SetProductSuppliersHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ProductSupplier as ProductSupplierDTO;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ProductSupplierDeleterInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ValueObject\ProductSupplierId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Exception\SupplierException;
+use ProductSupplier;
+
+/**
+ * Handles @var SetProductSuppliersCommand using legacy object model
+ */
+final class SetProductSuppliersHandler implements SetProductSuppliersHandlerInterface
+{
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @var ProductProvider
+     */
+    private $productProvider;
+
+    /**
+     * @var ProductSupplierDeleterInterface
+     */
+    private $productSupplierDeleter;
+
+    /**
+     * @var ProductSupplierPersister
+     */
+    private $productSupplierPersister;
+
+    /**
+     * @var ProductSupplierProvider
+     */
+    private $productSupplierProvider;
+
+    /**
+     * @param ProductUpdater $productUpdater
+     * @param ProductProvider $productProvider
+     * @param ProductSupplierDeleterInterface $productSupplierDeleter
+     * @param ProductSupplierPersister $productSupplierPersister
+     * @param ProductSupplierProvider $productSupplierProvider
+     */
+    public function __construct(
+        ProductUpdater $productUpdater,
+        ProductProvider $productProvider,
+        ProductSupplierDeleterInterface $productSupplierDeleter,
+        ProductSupplierPersister $productSupplierPersister,
+        ProductSupplierProvider $productSupplierProvider
+    ) {
+        $this->productUpdater = $productUpdater;
+        $this->productProvider = $productProvider;
+        $this->productSupplierDeleter = $productSupplierDeleter;
+        $this->productSupplierPersister = $productSupplierPersister;
+        $this->productSupplierProvider = $productSupplierProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SetProductSuppliersCommand $command): array
+    {
+        $productId = $command->getProductId();
+        $product = $this->productProvider->get($productId);
+        $defaultSupplierIdValue = $command->getDefaultSupplierId()->getValue();
+        $this->assertDefaultSupplierIsOneOfProvidedSuppliers($command);
+
+        $this->setProductSuppliers($productId, $command->getProductSuppliers());
+        $this->productUpdater->updateProductDefaultSupplier($product, $defaultSupplierIdValue);
+
+        return $this->getProductSupplierIds($productId);
+    }
+
+    /**
+     * @param SetProductSuppliersCommand $command
+     *
+     * @throws CannotUpdateProductException
+     */
+    private function assertDefaultSupplierIsOneOfProvidedSuppliers(SetProductSuppliersCommand $command): void
```

Comment:
I wonder if this should be in the Handler or in a dedicated service so that the domain logic is managed by the handler any more We talked about dedicated services per domain like a `ProductSupplierUpdater` in this case which can handle product relationship and sub domain constraints like these ones

Reply:
Or it could be managed by `ProductUpdater::updateProductDefaultSupplier` which might make more sense, it would just need to be called after all the update/creation/removals have been made (which is already the case)

Reply:
then we introduce a possibility for a non existing default supplier :thinking: (when updatable product suppliers list is missing current default supplier, we will update product suppliers, but default supplier will not be updated and will contain of supplier id which doesnt exist in product_supplier table anymore)).
To aviud this we could reset the default supplier to 0 if the default supplier update fails ? :thinking: